test/integration/model/findAll.test.js

```diff
@@ -1389,4 +1389,42 @@ describe(Support.getTestDialectTeaser('Model'), function() {
       expect(spy.called).to.be.ok;
     });
   });
+
+  describe('kenophobic mode', function() {
+    it('works from model options', function() {
+      var Model = current.define('Test', {
+        username: Sequelize.STRING(100)
+      },{
+        kenophobic: true
+      });
+
+      return Model.sync({ force: true })
+        .then(function() {
+          return expect(Model.findAll({
+            where: {
+              username: 'some-username-that-is-not-used-anywhere'
+            }
+          })).to.eventually.be.rejectedWith(Sequelize.EmptyResultError);
+        });
+    });
+
+    it('throws custom error', function() {
+
+      var Model = current.define('Test', {
+        username: Sequelize.STRING(100)
+      },{
+        kenophobic: new Sequelize.ConnectionError('Some Error') //using custom error instance
+      });
+
+      return Model.sync({ force: true })
+        .then(function() {
+          return expect(Model.findAll({
+            where: {
+              username: 'some-username-that-is-not-used-anywhere-for-sure-this-time'
+            }
+          })).to.eventually.be.rejectedWith(Sequelize.ConnectionError);
+        });
+    });
+  });
```

Comment:
We need an extra unit test so that we handle all 3 cases, none passed, class passed or instance passed